packages/core/src/embeddings/types.ts

```diff
@@ -10,15 +10,55 @@ export enum SimilarityType {
   EUCLIDEAN = "euclidean",
 }
 
+export type Embedding = number[];
+
+const DEFAULT_EMBED_BATCH_SIZE = 10;
+
 export abstract class BaseEmbedding {
+  embedBatchSize = DEFAULT_EMBED_BATCH_SIZE;
+
   similarity(
-    embedding1: number[],
-    embedding2: number[],
+    embedding1: Embedding,
+    embedding2: Embedding,
     mode: SimilarityType = SimilarityType.DEFAULT,
   ): number {
     return similarity(embedding1, embedding2, mode);
   }
 
-  abstract getTextEmbedding(text: string): Promise<number[]>;
-  abstract getQueryEmbedding(query: string): Promise<number[]>;
+  abstract getTextEmbedding(text: string): Promise<Embedding>;
+  abstract getQueryEmbedding(query: string): Promise<Embedding>;
+
+  async getTextEmbeddingBatch(
+    texts: string[],
+    options?: {
+      logProgress?: boolean;
+    },
+  ): Promise<Array<Embedding>> {
+    const resultEmbeddings: Array<Embedding> = [];
+    const chunkSize = this.embedBatchSize;
+
+    const queue: string[] = texts;
+    const totalChunks = Math.ceil(queue.length / chunkSize);
+
+    const processChunk = async (chunk: string[], chunkIndex: number) => {
+      const embeddings = await Promise.all(
+        chunk.map(async (text) => {
+          return await this.getTextEmbedding(text);
+        }),
+      );
+
+      resultEmbeddings.push(...embeddings);
```

Comment:
factor out this part (without the logging, can be kept here) as `protected processChunk` on the class level (see comment above)

this will then be the default chunk processor unless a subclass overrides the behavior